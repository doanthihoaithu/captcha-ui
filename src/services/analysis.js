import { requestServices } from 'services';
import * as constants from 'constants/index'


const analyzeQuery = (accessToken, data) => requestServices.customAxios.post(
    constants.analysis_analyze_query,
    data,
    {
        headers: {
            'Authorization': `Bearer ${accessToken}`,
            'Content-Type': 'multipart/form-data'
        }
    }
)

const generateQuery = (data) => requestServices.customAxios.post(
    constants.generate_captcha_query,
    data,
    {
        headers: {
            // 'Authorization': `Bearer ${accessToken}`,
            'Content-Type': 'multipart/form-data',
            // 'Content-Type': 'application/json'
        }
    }
)

export default {
    analyzeQuery,
    generateQuery
}