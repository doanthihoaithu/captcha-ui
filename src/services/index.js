// @flow
export { default as userServices } from './users';
export { default as requestServices } from './requests';
export { default as rootReducer } from './rootReducer';
export { default as store } from './store'
