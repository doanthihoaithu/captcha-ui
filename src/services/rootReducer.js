import { combineReducers } from 'redux';
import userReducer from './redux/user/reducers';
import statusReducer from './redux/status/reducers';

export default combineReducers({
	user: userReducer,
	status: statusReducer
});