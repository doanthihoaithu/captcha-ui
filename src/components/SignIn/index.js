import animationTrigger from 'helpers/loadingAnimationTrigger';
import moment from 'moment';
import React, { useRef, useState } from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { userServices } from 'services';
import Cookies from 'universal-cookie';
import * as actions from "../../services/redux/user/actions";
import { toast } from 'react-toastify'

const cookies = new Cookies();

export default () => {
    const username = useRef(null);
    const password = useRef(null);
    const dispatch = useDispatch();

    const [wrongAuthentication, updateWrongAuthentication] = useState(false);

    // const isAuthenticated = useSelector(state => state.user.isAuthenticated);

    function signUserIn(username, password) {
        // console.log(_username, _password)
        animationTrigger.startLoading();
        let data = { username, password }
        userServices.signUserIn(data).then((res) => {
            // console.log(res.data);
            cookies.set('leave', {
                username: res.data.username,
                access_token: res.data.access_token,
                refresh_token: res.data.refresh_token
            }, {
                    expires: new Date(moment().add('1800', 'seconds'))
                })
            updateWrongAuthentication(false)
            dispatch(actions.logIn(res.data))
            animationTrigger.stopLoading();
            toast.success('Login successfully')
        }).catch(err => {
            console.log(err)
            updateWrongAuthentication(true)
            animationTrigger.stopLoading();
        })
    }

    function onFormSubmit(event) {
        event.preventDefault();

        // console.log(event.target)
        const valid = event.target.checkValidity();
        // console.log(valid)
        if (valid) {
            signUserIn(username.current.value, password.current.value);
        }
    }

    return (
        <div className="container-contact100 animated fadeIn fast ">
            <div className="wrap-contact100">
                <button className="btn-hide-contact100">
                    <i className="zmdi zmdi-close"></i>
                </button>

                <div className="contact100-form-title" style={{ backgroundImage: `url(${require('./../../asset/images/bg-02.jpg')})` }}>
                    <span>Sign In</span>
                </div>

                <form className="contact100-form validate-form" onSubmit={onFormSubmit}>
                    <div className="wrap-input100 validate-input">
                        <input pattern="^(?=.{6,150}$)[a-zA-Z0-9]+$" type="text" ref={username} className="input100" id="exampleInputUsername1" aria-describedby="usernameHelp" placeholder="Enter username" required />
                        <span className="focus-input100"></span>
                        <label className="label-input100" htmlFor="username">
                            <span className="lnr lnr-user m-b-2"></span>
                        </label>
                    </div>


                    <div className="wrap-input100 validate-input">
                        <input pattern="^(?=.{8,150}$)[a-zA-Z0-9]+$" type="password" ref={password} className="input100" id="exampleInputPassword1" placeholder="Password" required />
                        <span className="focus-input100"></span>
                        <label className="label-input100" htmlFor="password">
                            <span className="lnr lnr-lock m-b-5"></span>
                        </label>
                    </div>

                    <div className="text-danger w-100 text-center mb-1" hidden={!wrongAuthentication}>Incorrect username or password</div>

                    <div className="container-contact100-form-btn">
                        <button className="contact100-form-btn" type="submit">
                            Sign In
    				    </button>
                    </div>
                </form>
                <div className="d-flex flex-column justify-content-center align-items-center mb-4">
                    <Link to="/forget-password" className="text-info">Forgot your password?</Link>
                    <div className="p-1">
                        <small>Don't have an account? </small><Link to="/signup" className="text-info">Create now</Link>
                    </div>
                </div>
            </div>
        </div>
    )
};
