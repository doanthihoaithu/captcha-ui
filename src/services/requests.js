// @flow
import axios from 'axios';
import * as actions from 'services/redux/user/actions';
import { store } from 'services'
import { toast } from 'react-toastify'

export const BASE_API_URL: string = `${process.env.REACT_APP_BASE_API_URL || 'localhost:3000'}`;

const customAxios = axios.create({
  baseURL: BASE_API_URL,
});

customAxios.interceptors.response.use(
  response => response,
  error => {
    // Add logic to handle error here
    // console.log(error.response.data)
    if (error.response && error.response.data.code === 500) {
      switch (error.response.data.message) {
        case "Signature has expired":
          store.dispatch(actions.logOut());
          toast.error('Your session has expired. Please sign in to continue');
          return Promise.reject(error);
        default:
          return Promise.reject(error);
      }
    }

    return Promise.reject(error);
  },
);

export default { customAxios };
