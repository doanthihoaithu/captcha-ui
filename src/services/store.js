import { createStore } from 'redux';
import { rootReducer } from './../services';

const store = createStore(rootReducer);

export default store;
