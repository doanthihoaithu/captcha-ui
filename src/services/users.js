// @flow
import { requestServices } from 'services';
import * as constants from 'constants/index'

// const fetchUsers = () => requestServices.customAxios.get('/').then(res => res.data);

const signUserIn = (data) => requestServices.customAxios.post(
    constants.user_authenticate_address,
    data,
    {
        header: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*',
        }
    }
)

const signUserOut = (accessToken) => requestServices.customAxios.post(
    constants.user_sign_out_address,
    {},
    {
        headers: {
            'Authorization': `Bearer ${accessToken}`
        }
    }
)

const registerNewUser = (data) => requestServices.customAxios.post(
    constants.user_sign_up_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const resetPassword = (data) => requestServices.customAxios.post(
    constants.user_forget_password_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const changePassword = (data, accessToken) => requestServices.customAxios.post(
    constants.user_change_password_address,
    data,
    {
        headers: {
            'Content-Type': 'application/json',
            'Authorization': `Bearer ${accessToken}`
            // 'Access-Control-Allow-Origin': '*'
        }
    }
)

const updateAvatar = (username, data, accessToken) => requestServices.customAxios.post(
    constants.user_update_avatar_address + username,
    data,
    {
        headers: {
            'Content-Type': 'multipart/form-data',
            'Authorization': `Bearer ${accessToken}`
        }
    }
)

const getAvatar = (username) => requestServices.customAxios.get(
    constants.user_get_avatar_address + username,
)

export default {
    signUserIn,
    signUserOut,
    registerNewUser,
    resetPassword,
    changePassword,
    updateAvatar,
    getAvatar,
};
