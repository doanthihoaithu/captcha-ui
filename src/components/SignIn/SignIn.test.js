import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import SignIn from '.';
import { store, userServices } from './../../services';


let component;
let signUserInAPI;

beforeEach(() => {
    component = mount(
        <Router>
            <Provider store={store}>
                <SignIn />
            </Provider>
        </Router>
    )
    signUserInAPI = jest.spyOn(userServices, 'signUserIn').mockReturnValue(Promise.resolve("value"));
})

afterEach(() => {
    signUserInAPI.mockReset();
    component.unmount();
})

it('should match snapshot', () => {
    expect(component).toMatchSnapshot()
})


it('should not call function signUserIn when submit nothing', () => {
    let signUserInAPI = jest.spyOn(userServices, 'signUserIn');

    let signUserInForm = component.find("form");

    signUserInForm.simulate('submit');

    expect(signUserInAPI).toHaveBeenCalledTimes(0);
})

it('should call function signUserIn when submit valid data', async () => {
    let signUserInForm = component.find("form");

    component.find("input[id='exampleInputUsername1']").getDOMNode().value = ("username")
    component.find("input[id='exampleInputPassword1']").getDOMNode().value = ("newpassword")

    component.update()

    expect(component.find("input")).toHaveLength(2)
    await signUserInForm.simulate('submit');

    expect(signUserInAPI).toHaveBeenCalledTimes(1);
})

it('should not call function signUserIn when submit invalid data', async () => {
    let signUserInForm = component.find("form");

    component.find("input[id='exampleInputUsername1']").getDOMNode().value = ("")
    component.find("input[id='exampleInputPassword1']").getDOMNode().value = ("newpassword")

    component.update()

    expect(component.find("input")).toHaveLength(2)
    await signUserInForm.simulate('submit');

    expect(signUserInAPI).toHaveBeenCalledTimes(0);
})