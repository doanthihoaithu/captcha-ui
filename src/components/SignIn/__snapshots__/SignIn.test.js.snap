// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <_default>
        <div
          className="container-contact100 animated fadeIn fast "
        >
          <div
            className="wrap-contact100"
          >
            <button
              className="btn-hide-contact100"
            >
              <i
                className="zmdi zmdi-close"
              />
            </button>
            <div
              className="contact100-form-title"
              style={
                Object {
                  "backgroundImage": "url(bg-02.jpg)",
                }
              }
            >
              <span>
                Sign In
              </span>
            </div>
            <form
              className="contact100-form validate-form"
              onSubmit={[Function]}
            >
              <div
                className="wrap-input100 validate-input"
              >
                <input
                  aria-describedby="usernameHelp"
                  className="input100"
                  id="exampleInputUsername1"
                  pattern="^(?=.{6,150}$)[a-zA-Z0-9]+$"
                  placeholder="Enter username"
                  required={true}
                  type="text"
                />
                <span
                  className="focus-input100"
                />
                <label
                  className="label-input100"
                  htmlFor="username"
                >
                  <span
                    className="lnr lnr-user m-b-2"
                  />
                </label>
              </div>
              <div
                className="wrap-input100 validate-input"
              >
                <input
                  className="input100"
                  id="exampleInputPassword1"
                  pattern="^(?=.{8,150}$)[a-zA-Z0-9]+$"
                  placeholder="Password"
                  required={true}
                  type="password"
                />
                <span
                  className="focus-input100"
                />
                <label
                  className="label-input100"
                  htmlFor="password"
                >
                  <span
                    className="lnr lnr-lock m-b-5"
                  />
                </label>
              </div>
              <div
                className="text-danger w-100 text-center mb-1"
                hidden={true}
              >
                Incorrect username or password
              </div>
              <div
                className="container-contact100-form-btn"
              >
                <button
                  className="contact100-form-btn"
                  type="submit"
                >
                  Sign In
                </button>
              </div>
            </form>
            <div
              className="d-flex flex-column justify-content-center align-items-center mb-4"
            >
              <Link
                className="text-info"
                to="/forget-password"
              >
                <a
                  className="text-info"
                  href="/forget-password"
                  onClick={[Function]}
                >
                  Forgot your password?
                </a>
              </Link>
              <div
                className="p-1"
              >
                <small>
                  Don't have an account? 
                </small>
                <Link
                  className="text-info"
                  to="/signup"
                >
                  <a
                    className="text-info"
                    href="/signup"
                    onClick={[Function]}
                  >
                    Create now
                  </a>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </_default>
    </Provider>
  </Router>
</BrowserRouter>
`;
